### PR TITLE
Improve auto expiration hint for trashbin and file versions

### LINKED ISSUE #27496: "No expiry configured." when AUTO is configured
The message from occ versions:expire is confusing:
```
# ./occ versions:expire
No expiry configured.
```
versions_retention_obligation is not configured, the default 'auto' is used then. See https://github.com/owncloud/core/issues/28509. I tested with NC 20.0.10 and 21.0.2.

### PR DESCRIPTION
**Pull request by szaimen**
Fix #27496

Signed-off-by: szaimen <szaimen@e.mail.de>

### CODE REVIEW

**Comment by szaimen** [comment]
Failed test is unrelated, afaics

**Comment by artonge** [comment]
Does this means that files will always expire and there is no way to keep the files in the trash for ever ?

**Comment by szaimen** [comment]
> Does this mean that files will always expire and there is no way to keep the files in the trash forever?

Actually, I don't know. lets cc @PVince81 and @icewind1991 on this

File: apps/files_trashbin/lib/Command/ExpireTrash.php [hunk 1]
```diff
@@ -74,7 +74,7 @@ protected function configure() {
 	protected function execute(InputInterface $input, OutputInterface $output): int {
 		$maxAge = $this->expiration->getMaxAgeAsTimestamp();
 		if (!$maxAge) {
-			$output->writeln("No expiry configured.");
+			$output->writeln("Auto expiration is configured - expiration will be handled automatically.");
```
Comment (Pytal) [inline apps/files_trashbin/lib/Command/ExpireTrash.php:77]:
```suggestion
			$output->writeln("Auto expiration is configured - keeps files and folders in the trash bin for 30 days and automatically deletes anytime after that if space is needed (note: files may not be deleted if space is not needed)");
```
As described here https://docs.nextcloud.com/server/latest/admin_manual/configuration_server/config_sample_php_parameters.html#deleted-items-trash-bin :)

File: apps/files_versions/lib/Command/ExpireVersions.php [hunk 2]
```diff
@@ -73,7 +73,7 @@ protected function configure() {
 	protected function execute(InputInterface $input, OutputInterface $output): int {
 		$maxAge = $this->expiration->getMaxAgeAsTimestamp();
 		if (!$maxAge) {
-			$output->writeln("No expiry configured.");
+			$output->writeln("Auto expiration is configured - expiration will be handled automatically.");
```
Comment (Pytal) [inline apps/files_versions/lib/Command/ExpireVersions.php:76]:
```suggestion
			$output->writeln("Auto expiration is configured - expiration will be handled automatically according to the expiration patterns detailed at the following link https://docs.nextcloud.com/server/latest/admin_manual/configuration_files/file_versioning.html.");
```
Mentioned link https://docs.nextcloud.com/server/latest/admin_manual/configuration_files/file_versioning.html :)

**Review by Pytal — CHANGES_REQUESTED** [review]

**Comment by Pytal** [comment]
> Does this mean that files will always expire and there is no way to keep the files in the trash forever?

Users may disable expiration https://docs.nextcloud.com/server/latest/admin_manual/configuration_server/config_sample_php_parameters.html#:~:text=space%20is%20needed)-,disabled,auto%20clean%20disabled%2C%20files%20and%20folders%20will%20be%20kept%20forever,-Defaults%20to%20auto :)

File: apps/files_trashbin/lib/Command/ExpireTrash.php (hunk 1 above)
Comment (artonge) [reply apps/files_trashbin/lib/Command/ExpireTrash.php:77]: Much clearer :+1:

Comment (acsfer) [reply apps/files_trashbin/lib/Command/ExpireTrash.php:77]: Agree

Comment (szaimen) [reply apps/files_trashbin/lib/Command/ExpireTrash.php]: done

File: apps/files_versions/lib/Command/ExpireVersions.php (hunk 2 above)
Comment (szaimen) [reply apps/files_versions/lib/Command/ExpireVersions.php]: done

**Comment by szaimen** [comment]
Thanks for the good suggestions @Pytal :)

**Comment by szaimen** [comment]
Failing test is unelated

**Comment by PVince81** [comment]
> > Does this mean that files will always expire and there is no way to keep the files in the trash forever?
> 
> Actually, I don't know. lets cc @PVince81 and @icewind1991 on this

according to config.sample.php, it's possible to disable this with: `trashbin_retention_obligation` => 'disabled'.
not sure how those settings map to the CLI output

**Review by Pytal — APPROVED** [review]

**Comment by szaimen** [comment]
Failing test is unrelated

**Review by artonge — APPROVED** [review]

**Pull request merged**